src/widgets/settingspages/GeneralPageView.cpp

```diff
@@ -362,4 +375,27 @@ void GeneralPageView::updateNavigationHighlighting()
     }
 }
 
+void GeneralPageView::addToolTip(QWidget &widget, QString text)
+{
+    if (!text.isEmpty())
+    {
+        if (text.length() > MAX_TOOLTIP_LINE_LENGTH)
+        {
+            // match MAX_TOOLTIP_LINE_LENGTH characters, any remaining
+            // non-space, and then capture the following space for
+            // replacement with newline
+            QString maxLineLengthPattern =
+                QString(".{%1}\\S*\\K(\\s+)").arg(MAX_TOOLTIP_LINE_LENGTH);
+            text.replace(QRegularExpression(maxLineLengthPattern), "\n");
```

Comment:
The `QRegularExpression` could be made static to reduce the overhead of parsing the expression